pkg/registry/apis/provisioning/jobs/migrate/worker.go

```diff
@@ -183,9 +199,11 @@ func (w *MigrationWorker) Process(ctx context.Context, repo repository.Repositor
 	}
 
 	// Clear unified and allow writing
-	err = worker.wipeUnifiedAndSetMigratedFlag(ctx, w.storageStatus)
-	if err != nil {
-		return fmt.Errorf("unable to reset unified storage %w", err)
+	if worker.legacy != nil {
```

Comment:
this should use the variable mentioned above instead of using the `worker.legacy`.